server/parser.go

```diff
@@ -171,6 +171,8 @@ func (c *client) parse(buf []byte) error {
 						s.mu.Unlock()
 						if exists {
 							c.RegisterUser(user)
+							c.clearAuthTimer()
```

Comment:
Please surround this and the call below with `c.mu.Lock()/c.mu.Unlock()`, since those two require locking.